src/Stratis.Bitcoin.Features.Consensus/IBDStateProvider.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using NBitcoin;
+using Stratis.Bitcoin.Base;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus
+{
+    public class IBDStateProvider : IBlockDownloadState
```

Comment:
Can you describe the purpose of this class here? Lets not introduce new class without documentation.

Reply:
Also, lets rename this BlockDownloadStateProvider, so that it matches IBlockDownloadState? Either that or rename IBlockDownloadState?

Reply:
Fixed.